hibernate-core/src/test/java/org/hibernate/orm/test/sql/ast/TupleInListAlternativeTests.java

```diff
@@ -0,0 +1,149 @@
+})
+@ServiceRegistry
+@SessionFactory
+@RequiresDialect(DB2Dialect.class)
+public class TupleInListAlternativeTests {
+
+    @BeforeEach
+    public void setup(SessionFactoryScope scope) {
+        scope.inTransaction(
+                session -> {
+                    EntityWithEmbeddedSubEntity entity = new EntityWithEmbeddedSubEntity();
+                    entity.setId( 1 );
+                    entity.setSub( new EntityWithEmbeddedSubEntity.Sub( "ZH2", "HDS" ) );
+                    entity.setPresent( "Y" );
+                    session.persist( entity );
+                    EntityWithEmbeddedSubEntity entity2 = new EntityWithEmbeddedSubEntity();
+                    entity2.setId( 2 );
+                    entity2.setSub( new EntityWithEmbeddedSubEntity.Sub( "ZH3", "HDS" ) );
+                    entity2.setPresent( "Y" );
+                    session.persist( entity2 );
+                    EntityWithEmbeddedSubEntity entity3 = new EntityWithEmbeddedSubEntity();
+                    entity3.setId( 3 );
+                    entity3.setSub( new EntityWithEmbeddedSubEntity.Sub( "GXZ", "HDS" ) );
+                    entity3.setPresent( "N" );
+                    session.persist( entity3 );
+                    EntityWithEmbeddedSubEntity entity4 = new EntityWithEmbeddedSubEntity();
+                    entity4.setId( 4 );
+                    entity4.setSub( new EntityWithEmbeddedSubEntity.Sub( "KAZ", "TST" ) );
+                    entity4.setPresent( "Y" );
+                    session.persist( entity4 );
+                } );
+    }
+
+    @AfterEach
+    public void cleanup(SessionFactoryScope scope) {
+        scope.inTransaction(
+                session -> session.createMutationQuery( "delete from EntityWithEmbeddedSubEntity" ).executeUpdate() );
+    }
+
+    @Test
+    @Jira("https://hibernate.atlassian.net/browse/HHH-16886")
+    public void testQueryResultsForHhh16886(SessionFactoryScope scope) {
+        scope.inTransaction(
+                session -> {
+                    List<EntityWithEmbeddedSubEntity> resultList = session
+                            .createQuery( "select e from EntityWithEmbeddedSubEntity e where e.sub in :subs and e.present = :present",
+                                    EntityWithEmbeddedSubEntity.class )
+                            .setParameter( "subs",
+                                    Arrays.asList( new EntityWithEmbeddedSubEntity.Sub( "ZH2", "HDS" ), new EntityWithEmbeddedSubEntity.Sub( "GXZ", "HDS" ),
+                                            new EntityWithEmbeddedSubEntity.Sub( "KAZ", "TST" ) ) )
+                            .setParameter( "present", "Y" ).list();
+                    assertThat( resultList.size(), equalTo( 2 ) );
+                    assertThat( resultList.stream().map( EntityWithEmbeddedSubEntity::getId ).toList(), hasItems( 1, 4 ) );
+                    assertThat( resultList.stream().map( EntityWithEmbeddedSubEntity::getId ).toList(), not( hasItems( 2, 3 ) ) );
```

Comment:
This is causing a compilation error, note that the project is built with Java 11.

```suggestion
                    assertThat( resultList.stream().map( EntityWithEmbeddedSubEntity::getId ).collect( Collectors.toList() ), hasItems( 1, 4 ) );
                    assertThat( resultList.stream().map( EntityWithEmbeddedSubEntity::getId ).collect( Collectors.toList() ), not( hasItems( 2, 3 ) ) );
```

Reply:
thanks for the hints - it's fixed